Gems/Camera/Code/Source/EditorCameraComponent.cpp

```diff
@@ -193,11 +200,34 @@ namespace Camera
         return AZ::Edit::PropertyRefreshLevels::AttributesAndValues;
     }
 
+    AZ::Crc32 EditorCameraComponent::OnMatchViewportClicked()
+    {
+        if (IsThisCamera())
+        {
+            AZ_Warning("EditorCameraComponent", false, "Camera %s is already active.", GetEntity()->GetName().c_str());
+            return AZ::Edit::PropertyRefreshLevels::AttributesAndValues;
+        }
+        AZStd::optional<AZ::Transform> transform = AZStd::nullopt;
+        EditorCameraRequests::Bus::BroadcastResult(transform, &EditorCameraRequests::GetActiveCameraTransform);
+        if (!transform)
+        {
+            return AZ::Edit::PropertyRefreshLevels::AttributesAndValues;
+        }
+        AZStd::optional<float> fov = AZStd::nullopt;
+        EditorCameraRequests::Bus::BroadcastResult(fov, &EditorCameraRequests::GetCameraFoV);
+        if (!fov)
+        {
+            return AZ::Edit::PropertyRefreshLevels::AttributesAndValues;
+        }
+        AZ::TransformBus::Event(GetEntityId(), &AZ::TransformInterface::SetWorldTM, transform.value());
```

Comment:
nit: Instead of calling `GetEntityId()` twice you could add it as a local variable `const AZ::EntityId entityId = GetEntityId();`